imod/mf6/pkgbase.py

```diff
@@ -8,9 +9,42 @@
 
 import imod
 from imod.mf6.interfaces.ipackagebase import IPackageBase
+from imod.typing.grid import GridDataset
 
 TRANSPORT_PACKAGES = ("adv", "dsp", "ssm", "mst", "ist", "src")
 EXCHANGE_PACKAGES = "gwfgwf"
+ARGS_TO_EXCLUDE = ["validate"]
+
+
+def merge_unstructured_dataset(variables_to_merge):
+    """Work around xugrid issue https://github.com/Deltares/xugrid/issues/179"""
+
+    # Separate variables into list of grids and dict of scalar variables
+    grids_ls = []
+    scalar_dict = {}
+    for name, variable in variables_to_merge.items():
+        if isinstance(variable, xu.UgridDataArray):
+            grids_ls.append(variable.rename(name))
+        else:
+            scalar_dict[name] = variable
+
+    # Merge grids
+    dataset = xu.merge(grids_ls, join="exact")
+
+    # Assign scalar variables manually
+    for name, variable in scalar_dict.items():
+        dataset[name] = variable
+
+    return dataset
+
+
+def merge_at_init(variables_to_merge):
+    if variables_to_merge is not None:
+        for arg in variables_to_merge.values():
+            if isinstance(arg, xu.UgridDataArray):
+                return merge_unstructured_dataset(variables_to_merge)
```

Comment:
This check can be made a bit stricter. We don;t want to mix dataarrays and ugriddataarays

```
def merge_at_init(variables_to_merge):
    if variables_to_merge is None:
        return xr.Dataset()

    if all(isinstance(variable, xr.Dataarray) for variable in variables_to_merge):
        return xr.merge([variables_to_merge], join="exact")
    
    if all(isinstance(variable, xu.UgridDataArray) for variable in variables_to_merge):
        return merge_unstructured_dataset(variables_to_merge)
    
    raise ArgumentError("Mixed GridDataArray types. Unable to merge")
```